src/app/static/src/app/utils.ts

```diff
@@ -411,8 +421,35 @@ export const constructRehydrateProjectState = (rootState: RootState, data: Proje
         projects,
         baseline,
         surveyAndProgram,
-        stepper
+        modelOptions,
+        stepper,
+        hintrVersion: {
+            hintrVersion: data.state.version
+        }
     }
 
     return {files, savedState}
 }
+
+export const constructOptionsFormMetaFromData = (state: ModelOptionsState, meta: DynamicFormMeta) => {
+    const payloadContainsMeta = meta.controlSections.length > 0
+    const stateContainsOptions = Object.keys(state.options).length > 0
+    const emptyStateMeta = state.optionsFormMeta.controlSections.length === 0
```

Comment:
You'll be able to take this check out here if you only call this when this is the case, as suggested above